src/components/Menu/MenuItem.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import {ListItem} from '~/components/ListItem';
+import classnames from 'clsx';
+import './MenuItem.scss';
+
+type TMenuItemVariant = 'default' | 'title';
+enum MenuItemVariant {
+    DEFAULT = 'default',
+    TITLE = 'title'
+}
+
+type TMenuItemImageSize = 'small' | 'big';
+enum MenuItemImageSize {
+    SMALL = 'small',
+    BIG = 'big'
+}
+
+interface IMenuItemProps {
+    /**
+     * Additional classname
+     */
+    className?: string,
+
+    /**
+     * Is item being hovered
+     */
+    isHover?: boolean,
+
+    /**
+     * Is item selected
+     */
+    isSelected?: boolean,
+
+    /**
+     * Is item disabled
+     */
+    isDisabled?: boolean,
+
+    /**
+     * MenuItem label
+     */
+    label: React.ReactNode,
+
+    /**
+     * Does the label contain HTML markup
+     */
+    isHtml?: boolean,
+
+    /**
+     * Icon display before the label
+     */
+    iconStart?: React.ReactElement,
+
+    /**
+     * Icon display at the end of MenuItem
+     */
+    iconEnd?: React.ReactElement,
+
+    /**
+     * MenuItem variants
+     */
+    variant?: TMenuItemVariant,
+
+    /**
+     * Optional image to display to describe the menu item
+     */
+    image?: HTMLImageElement,
```

Comment:
That could be okay - the reason I was thinking to use an `image` element is so that the user could provide alt text for the image. However, we could just use the label provided for the alt text as well.

Reply:
Good point, we can keep it like this